### PR TITLE
Add networking docs

### PR DESCRIPTION
**Pull request by mit-mit**
Adding a networking topic based on https://github.com/dart-lang/http/pull/331

### CODE REVIEW

**Comment by legalcodes** [comment]
LGTM

File: src/docs/development/data-and-backend/networking.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+```
+
+## Samples
+
+For a practical samples of various networking tasks (incl. fetching data,
+WebSockets, and parsing data in the background) see the 
+[networking cookbook(https://flutter.dev/docs/cookbook#networking).
+
+[`http`]: {{site.pub-pkg}}/http
+[declare]: https://developer.android.com/training/basics/network-ops/connecting
```
Comment (legalcodes) [inline src/docs/development/data-and-backend/networking.md]: Nit:

```suggestion
[declare]: {{site.android-dev}}/training/basics/network-ops/connecting
```

File: src/docs/development/data-and-backend/networking.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+title: Networking
+description: Internet network calls in Flutter.
+---
+
+## Cross-platform http networking
+
+The [`http`][] package provides the simplest way to issue http requests. This
+package is supported on Android, iOS, and the web.
+
+## Platform notes
```
Comment (legalcodes) [inline src/docs/development/data-and-backend/networking.md]: Nit: Don't use empty headings or headings with no associated content.

https://developers.google.com/style/headings

File: src/docs/development/data-and-backend/networking.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+<manifest xlmns:android...>
+ ...
+ <uses-permission android:name="android.permission.INTERNET" />
+ <application ...
+</manifest>
+```
+
+## Samples
+
+For a practical samples of various networking tasks (incl. fetching data,
```
Comment (sfshaza2) [inline src/docs/development/data-and-backend/networking.md]: In "For a practical samples...",   "samples" => "sample".

File: src/docs/development/data-and-backend/networking.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+WebSockets, and parsing data in the background) see the 
+[networking cookbook(https://flutter.dev/docs/cookbook#networking).
```
Comment (sfshaza2) [inline src/docs/development/data-and-backend/networking.md]: Change URL to a relative link: "/docs/cookbook#networking". Also, we generally put the urls at the end of the file, in alphabetical order.

Comment (legalcodes) [inline src/docs/development/data-and-backend/networking.md]: I also just noticed this line is missing a closing square bracket:

```suggestion
[networking cookbook](https://flutter.dev/docs/cookbook#networking).
```

File: src/docs/development/data-and-backend/networking.md (hunk 2 above)
Comment (mit-mit) [reply src/docs/development/data-and-backend/networking.md]: Good point, clarified

**Pull request merged**